### PR TITLE
AD-13759: Bluescape cli to support legacy enterprise license provisioning

### PR DESCRIPTION
**Pull request by Rakshitha-Sheshagiri**

### CODE REVIEW

**Comment by Rakshitha-Sheshagiri** [comment]
Screenshot from 2022-11-18 11-30-01

File: src/commands/provisionlicense/link-external-legacy-subscription.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      'licensesCurrentlyInUse',
+      'organizationStorageLimitMb',
+      'mode',
+      'planName',
+      'updatedAt',
+    ],
+  );
+
+  if (result.errors) {
+    spinner.fail(chalk.red(result.errors[0].message));
```
Comment (HMageswari) [inline src/commands/provisionlicense/link-external-legacy-subscription.ts:87]: Use a valid check for object properties, what happens if it is not present.

File: src/commands/provisionlicense/ask-provision-license-information.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import inquirer from 'inquirer';
+import { isExternalSubscriptionId, isNumber } from '../../utils/validators';
+
+export async function askLegacySubscriptionDetails(): Promise<any> {
+  const legacySubscriptionInputPrompt = [
+    {
+      name: 'externalSubscriptionId',
+      type: 'input',
+      message: 'Enter the external subscription Id:',
```
Comment (HMageswari) [inline src/commands/provisionlicense/ask-provision-license-information.ts:9]: Please make sure to test the negative case

Comment (Rakshitha-Sheshagiri) [reply src/commands/provisionlicense/ask-provision-license-information.ts:9]: added the validation for subscription id where the provided value should be a string with the length between 2 to 50

File: src/commands/provisionlicense/link-external-legacy-subscription.ts (hunk 1 above)
Comment (Rakshitha-Sheshagiri) [reply src/commands/provisionlicense/link-external-legacy-subscription.ts]: added if condition with valid checks

**Review by HMageswari — APPROVED** [review]

**Review by ssethu24 — APPROVED** [review]

**Comment by ssethu24** [comment]
@Rakshitha-Sheshagiri  Please update readme file

**Pull request merged**